libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1334,28 +1354,110 @@ string YulUtilFunctions::allocationFunction()
 	});
 }
 
-string YulUtilFunctions::allocateMemoryArrayFunction(ArrayType const& _type)
+string YulUtilFunctions::zeroMemoryArrayFunction(ArrayType const& _type)
 {
-	solUnimplementedAssert(!_type.isByteArray(), "");
+	if (_type.baseType()->hasSimpleZeroValueInMemory())
+		return zeroSimpleMemoryArrayFunction(_type);
+	return zeroComplexMemoryArrayFunction(_type);
+}
 
-	string functionName = "allocate_memory_array_" + _type.identifier();
+string YulUtilFunctions::zeroSimpleMemoryArrayFunction(ArrayType const& _type)
+{
+	solAssert(_type.baseType()->hasSimpleZeroValueInMemory(), "");
+
+	string functionName = "zero_simple_memory_array_" + _type.identifier();
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
-			function <functionName>(length) -> memPtr {
-				memPtr := <alloc>(<allocSize>(length))
-				<?dynamic>
-				mstore(memPtr, length)
-				</dynamic>
+			function <functionName>(dataStart, dataSizeInBytes) {
```

Comment:
This function is not related to arrays anymore at all :)

Reply:
Haha indeed ;)